src/core/init/duration_updater.ts

```diff
@@ -42,25 +63,34 @@ export default function DurationUpdater(
 ) : Observable<never> {
   return fromEvent(manifest, "manifestUpdate").pipe(
     startWith(null),
-    tap(() => {
-      const maximumPosition = manifest.getMaximumPosition();
-
-      // Some targets poorly support setting a very high number for durations.
-      // Yet, in live contents, we would prefer setting a value as high as possible
-      // to still be able to seek anywhere we want to (even ahead of the Manifest if
-      // we want to). As such, we put it at a safe default value of 2^32 excepted
-      // when the maximum position is already relatively close to that value, where
-      // we authorize exceptionally going over it.
-      const newDuration = !manifest.isLive ? maximumPosition :
-                                             Math.max(Math.pow(2, 32),
-                                                      maximumPosition + YEAR_IN_SECONDS);
+    scan((acc) => {
+      return [acc[1], manifest.getMaximumPosition()] as [number, number];
+    }, [null, manifest.getMaximumPosition()] as [null | number, number]),
```

Comment:
I'm under the impression that it's not emitting the first time (before the first `manifestUpdate`).

Also wouldn't it be cleaner to use something like pairwise?
And what about just having the previous value as a variable there, would it be problematic? I'm not sure (at worst in an observableDefer(()).

Reply:
Now that I re-read it, it emits two times the same value the first time, is that it?
It's not very elegant.

Reply:
Why did you resolve that conversation?

Reply:
I pushed again, and it seems that it does not emit twice anymore. Or am i wrong ?

Reply:
It doesn't emit twice, but it emits the same value two times in a tuple the first time (subtle difference :p), no?

Also why not using something like pairwise, it could be more readable with it (or is it more complicated?)